subsys/bluetooth/controller/ll_sw/ull_sched.c

```diff
@@ -596,19 +552,161 @@ static void win_offset_calc(struct ll_conn *conn_curr, uint8_t is_select,
 	*offset_max = offset_index;
 }
 #endif /* CONFIG_BT_LL_SW_LLCP_LEGACY */
-
 #endif /* CONFIG_BT_CTLR_CONN_PARAM_REQ */
+#endif /* CONFIG_BT_CONN */
+
+static uint8_t after_xxxx_slot_get(uint8_t user_id, uint32_t ticks_slot_abs,
+				   ticker_op_match_func ticker_match_op_cb,
+				   ull_hdr_get_func ull_hdr_get_cb,
+				   uint32_t *ticks_anchor,
+				   uint32_t *ticks_to_expire_xxxx,
+				   uint32_t *ticks_slot_xxxx)
+{
+	uint32_t ticks_to_expire_prev;
+	uint32_t ticks_slot_abs_prev;
+	uint32_t ticks_to_expire;
+	uint8_t ticker_id_prev;
+	uint8_t ticker_id;
+
+	ticks_slot_abs += HAL_TICKER_US_TO_TICKS(EVENT_JITTER_US << 3);
+
+	ticker_id = ticker_id_prev = 0xff;
+	ticks_to_expire = ticks_to_expire_prev = 0U;
+	ticks_slot_abs_prev = 0U;
+	while (1) {
+		uint32_t ticks_slot_abs_curr = 0;
+		uint32_t ticks_to_expire_normal;
+		uint32_t volatile ret_cb;
+		struct ull_hdr *hdr;
+		uint32_t ret;
+		bool success;
+
+		ret_cb = TICKER_STATUS_BUSY;
+#if defined(CONFIG_BT_TICKER_NEXT_SLOT_GET_MATCH)
+		ret = ticker_next_slot_get_ext(TICKER_INSTANCE_ID_CTLR, user_id,
+					       &ticker_id, ticks_anchor,
+					       &ticks_to_expire,
+					       NULL, /* lazy */
+					       ticker_match_op_cb,
+					       NULL, /* match_op_context */
+					       ticker_op_cb, (void *)&ret_cb);
+#else /* !CONFIG_BT_TICKER_NEXT_SLOT_GET_MATCH */
+		ret = ticker_next_slot_get(TICKER_INSTANCE_ID_CTLR, user_id,
+					   &ticker_id, ticks_anchor,
+					   &ticks_to_expire,
+					   ticker_op_cb, (void *)&ret_cb);
+#endif /* !CONFIG_BT_TICKER_NEXT_SLOT_GET_MATCH */
+		if (ret == TICKER_STATUS_BUSY) {
+			while (ret_cb == TICKER_STATUS_BUSY) {
+				ticker_job_sched(TICKER_INSTANCE_ID_CTLR,
+						 user_id);
+			}
+		}
+
+		success = (ret_cb == TICKER_STATUS_SUCCESS);
+		LL_ASSERT(success);
```

Comment:
```suggestion
		LL_ASSERT(ret_cb == TICKER_STATUS_SUCCESS);
```

And remove the single use variable

Reply:
as `ret_cb` is volatile, it cannot be used inside `LL_ASSERT`, it is a coverity rule related to use of volatile variable that it think would change when being checked.

Reply:
Can you please add that as a comment then? That is useful information to know if anyone ever tries to modify the code :)

Reply:
comment added in code.